### PR TITLE
Bring `kategory-annotations` to the `kategory` repo

### PR DESCRIPTION
**Pull request by raulraja**
Also fixes a bunch of instances issues and provides some  `implicit` objects where instances can be looked up without the need to manually register anything.
- [x] `kategory-annotations` migrations
- [ ] Derive individual instances for all auto `@deriving` datatypes (will be addresed in #260)

### CODE REVIEW

File: kategory-test/src/main/kotlin/kategory/UnitSpec.kt [hunk 1]
```diff
@@ -8,29 +8,7 @@ import io.kotlintest.specs.StringSpec
  * Base class for unit tests
  */
 abstract class UnitSpec : StringSpec() {
-    companion object {
```
Comment (pakoito) [inline kategory-test/src/main/kotlin/kategory/UnitSpec.kt]: Ho ho ho hoooo

File: detekt.yml [hunk 2]
```diff
@@ -78,7 +80,7 @@ formatting:
     active: true
     autoCorrect: true
   OptionalUnit:
-    active: true
+    active: false
```
Comment (pakoito) [inline detekt.yml]: Ideally we'd need the opposite option: flagging places where Unit isn't there. For us explicitness is key and we've been bitten by returns that should be Unit instead before.

File: kategory-core/src/main/kotlin/kategory/data/ListKW.kt [hunk 3]
```diff
@@ -59,17 +59,17 @@ data class ListKW<out A> constructor(val list: List<A>) : ListKWKind<A>, List<A>
             return ListKW(buf)
         }
 
-        fun functor(): ListKWHKMonadInstance = ListKW.monad()
+        fun functor(): ListKWMonadInstance = ListKW.monad()
```
Comment (tonilopezmr) [inline kategory-core/src/main/kotlin/kategory/data/ListKW.kt]: Why haven't you deleted these methods too like in `Id` or `NonEmptyList`?

File: detekt.yml (hunk 2 above)
Comment (raulraja) [reply detekt.yml]: Agreed, If detekt rules can be user created we can also provide some to encourage a more functional style.

File: kategory-core/src/main/kotlin/kategory/data/ListKW.kt (hunk 3 above)
Comment (raulraja) [reply kategory-core/src/main/kotlin/kategory/data/ListKW.kt]: I though I did, but will look over before merging

**Pull request merged**